calico/felix/fetcd.py

```diff
@@ -659,6 +728,26 @@ def on_workload_delete(self, response, hostname, orchestrator,
             del self.endpoint_ids_per_host[hostname]
 
 
+def _reconnect(etcd_communicator, copy_cluster_id=True):
+    """ Parameter etcd_communicator can be either etcd_watcher or etcd_api"""
```

Comment:
function doctring should explain what the function does and then detail the params